### PR TITLE
Avoid needless VECTOR_size errors

### PR DESCRIPTION
**Pull request by damonbarry**
This is a follow-up to #124. There were a few more VECTOR_size errors that we can avoid by testing for `gateway_handle->links == NULL`.

@nielsms

### CODE REVIEW

File: core/src/gateway_internal.c [hunk 1]
```diff
@@ -690,23 +690,26 @@ int add_module_to_any_source(GATEWAY_HANDLE_DATA* gateway_handle, MODULE_DATA* m
...
-            if (module_sink == NULL)
+            LINK_DATA * link_data = VECTOR_element(gateway_handle->links, link);
+            if (link_data->from_any_source)
             {
-                LogError("Could not find sink for link [%s]", link_data->module_sink);
-            }
-            else
-            {
-                if (remove_one_link_from_broker(gateway_handle, module->module, (*module_sink)->module) != 0)
+                MODULE_DATA** module_sink = (MODULE_DATA**)VECTOR_find_if(gateway_handle->modules, module_name_find, link_data->module_sink->module_name);
```
Comment (aribeironovaes) [inline core/src/gateway_internal.c]: Do we need to test if link data is null?

Comment (damonbarry) [inline core/src/gateway_internal.c]: I don't think so. If `gateway_handle->links` is non-NULL, and VECTOR_size returns a non-zero value, then the vector element will be valid. We maintain this vector internally, so we can guarantee consistency/integrity.
